From dbf68c73961764083f826c315ae11bec8ebb3a3d Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Mon, 3 Oct 2022 13:50:36 +0100
Subject: [PATCH 09/28] fs_mgr: Keep allowing encryptable fstab flag

Some users still use "encryptable" flag while being unencrypted. Let them still boot their devices.

Change-Id: Id2554d6563a869bb5bcf39c67b0322484bdab0c9
---
 libfstab/fstab.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/libfstab/fstab.cpp b/libfstab/fstab.cpp
index e3e5336..4e32ffb 100644
--- a/libfstab/fstab.cpp
+++ b/libfstab/fstab.cpp
@@ -360,16 +360,6 @@ bool ParseFsMgrFlags(const std::string& flags, FstabEntry* entry) {
         }
     }

-    // FDE is no longer supported, so reject "encryptable" when used without
-    // "vold_managed".  For now skip this check when in recovery mode, since
-    // some recovery fstabs still contain the FDE options since they didn't do
-    // anything in recovery mode anyway (except possibly to cause the
-    // reservation of a crypto footer) and thus never got removed.
-    if (entry->fs_mgr_flags.crypt && !entry->fs_mgr_flags.vold_managed && !InRecovery()) {
-        LERROR << "FDE is no longer supported; 'encryptable' can only be used for adoptable "
-                  "storage";
-        return false;
-    }
     return true;
 }

--
2.55.0
